### PR TITLE
Don't have duplicated code in the docker-compose files

### PR DESCRIPTION
**Pull request by sbrunner**
By using the extends attribute that use a lib file

## Proposed change

Don't have duplicated code in the docker-compose files
By using the extends attribute that use a lib file

## Type of change

Less duplicated code

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Other Less duplicated code

## Checklist:

- [X] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [X] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [X] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [X] I have made corresponding changes to the documentation as needed.
- [X] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Review by shamoon — COMMENTED** [review]
This a cool thing compose can do but honestly I wonder if it’s really better for our users or if it will increase complexity. Currently a user can just look at one compose file on GitHub and copypasta or pull one file, rename and use it and things are pretty clear. IMHO this obfuscates things a bit and the “benefit” isn’t so high since these files are really just maintained as a starting point for users getting setup, so it’s not like the code duplication is relevant to the software itself.

I don’t feel super strongly about it, welcome other thoughts of course but I can’t help but wonder if it might be easier for the average user to leave them as-is

**Review by stumpylog — COMMENTED** [review]
I do see how this could add a layer of obfuscation that less experienced users might find confusing or intimidating.  But less locations to update versions is handy.  I'm not sure which outweighs what.

(Also this would break the install script as is)

**Comment by sbrunner** [comment]
My final idea is to be able to use the Docker compose files directly in Portainer, to be more explicit I create a draft pull request, see:
https://github.com/paperless-ngx/paperless-ngx/pull/2661

If you prefer, I can create a new set of docker compose files, for me i't not such a bad solution because I will be more free to introduce some changes to me more like it should for Portainer without take the rest to breaking the other users.

What do you think?

**Comment by sbrunner** [comment]
Finally, in #2661 I create another set of Docker compose, if you agree with that just close this pull request :-)

File: docs/setup.md [hunk 1]
```diff
@@ -882,3 +883,59 @@ reverse proxy. Please refer to the [hosting and security](/configuration#hosting
 Also read
 [this](https://channels.readthedocs.io/en/stable/deploying.html#nginx-supervisor-ubuntu),
 towards the end of the section.
+
+### Install on Protainer {#portainer}
```
Comment (jonaswinkler) [inline docs/setup.md:887]: typo!

Comment (shamoon) [reply docs/setup.md:887]: There are a few typos here :) but there’s a larger question about this. There are actually 3 PRs for the same question, would be great to have your thoughts! See https://github.com/paperless-ngx/paperless-ngx/pull/2663 for the latest discussion of it

**Pull request closed without merging**